sharding/xorValidatorShuffler.go

```diff
@@ -240,13 +258,21 @@ func shuffleOutNodes(
 	newEligible := make(map[uint32][]Validator)
 	var shardShuffledOut []Validator
 
-	for shard, validators := range eligible {
+	for shard := uint32(0); shard < uint32(len(eligible)-1); shard++ {
```

Comment:
it think it would have been easier to received [][]Validator instead of map[uint32]validators, this way a lot of code is duplicated, plus shuffler needs to know about metachainshardID.

Reply:
fixed